tools/adventure-pack/goodies/typescript/Iterator.prototype.every/index.test.ts

```diff
@@ -0,0 +1,53 @@
+import { describe, expect, it } from "@jest/globals";
+
+import "./index";
+
+describe("Iterator.prototype.every", () => {
+  it("returns true if all elements pass the test and false if an element fails a test", () => {
+    const isEven = (element: number) => element % 2 === 0;
+    expect([2, 4, 6, 8, 10].values().every(isEven)).toBe(true);
+    expect([2, 4, 6, 7, 8, 10].values().every(isEven)).toBe(false);
+    expect([1, 4, 6, 8, 10].values().every(isEven)).toBe(false);
+    expect([2, 4, 6, 8, 11].values().every(isEven)).toBe(false);
```

Comment:
When it's not too hard, try not to have obvious patterns in the test data -- for example if the elements are always sorted then the test is slightly weaker. What if due to a bug the implementation only worked on sorted input?

The strongest tests are super varied!